// vite-plugin-gh-pages/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ghPages = void 0;
const node_fs_1 = __importDefault(require("node:fs"));
const node_path_1 = __importDefault(require("node:path"));
const gh_pages_1 = __importDefault(require("gh-pages"));
const get_package_name_1 = require("./get-package-name");
const ghPages = (options) => {
    var _a, _b;
    let outDir = "";
    const onError = (_a = options === null || options === void 0 ? void 0 : options.onError) !== null && _a !== void 0 ? _a : ((error) => console.log(error));
    const onPublish = (_b = options === null || options === void 0 ? void 0 : options.onPublish) !== null && _b !== void 0 ? _b : (({ outDir, branch }) => {
        console.log(`🎉 Published \`${outDir}\` to branch \`${branch}\`.`);
    });
    return {
        name: "vite:gh-pages",
        apply: "build",
        enforce: "post",
        config(config) {
            if (config.base === undefined) {
                config.base = "/" + (0, get_package_name_1.getPackageName)() + "/";
            }
        },
        configResolved(resolvedConfig) {
            outDir = resolvedConfig.build.outDir;
        },
        closeBundle() {
            var _a;
            node_fs_1.default.writeFileSync(node_path_1.default.join(outDir, ".nojekyll"), "");
            const gpOptions = Object.assign({ dotfiles: true, branch: "gh-pages" }, options);
            (_a = options === null || options === void 0 ? void 0 : options.onBeforePublish) === null || _a === void 0 ? void 0 : _a.call(options, Object.assign(Object.assign({}, gpOptions), { outDir }));
            gh_pages_1.default.publish(outDir, gpOptions, (error) => {
                if (error)
                    return onError(error);
                onPublish(Object.assign(Object.assign({}, gpOptions), { outDir }));
            });
        },
    };
};
exports.ghPages = ghPages;
